fig/backends/aws/__init__.py

```diff
@@ -1,50 +1,153 @@
-import json
 from datetime import datetime
+import traceback
 import boto3
+from botocore.exceptions import ClientError
 from ...config import config
 from ...log import log
 
 
 class Submitter():
-    def __init__(self, queue, event):
-        self.sqs_queue = queue
+    def __init__(self, event):
         self.event = event
 
+    def find_instance(self, instance_id, mac_address):  # pylint: disable=R0201
+        # Instance IDs are unique to the region, not the account, so we have to check them all
+
+        ec2_client = boto3.client("ec2")
+        regions = [region["RegionName"] for region in ec2_client.describe_regions()["Regions"]]
+        for region in regions:
+            ec2 = boto3.resource("ec2", region_name=region)
+            try:
+                ec2instance = ec2.Instance(instance_id)
+                found = False
+                # Confirm the mac address matches
+                for iface in ec2instance.network_interfaces:
+                    det_mac = mac_address.lower().replace(":", "").replace("-", "")
+                    ins_mac = iface.mac_address.lower().replace(":", "").replace("-", "")
+                    if det_mac == ins_mac:
+                        found = True
+                if found:  # pylint: disable=R1723
+                    break
```

Comment:
Nitpick. If we `return` here instead of `break` we may end up with less code that is cleaner and has lower complexity.

Further, we may want to return `return None` at the bottom of the method instead of returning the `False`